packages/react/src/components/Tabs/Tabs.tsx

```diff
@@ -0,0 +1,165 @@
+  const focusedTabRef = useRef<HTMLLIElement>(null);
+
+  const tabs = React.Children.toArray(children).filter(
+    child => (child as React.ReactElement<any>).type === Tab
+  );
+  const panels = React.Children.toArray(children).filter(
+    child => (child as React.ReactElement<any>).type === TabPanel
+  );
+
+  const tabCount = tabs.length;
+
+  const handleClick = (event: React.MouseEvent) => {
+    const eventTarget: TargetElement = event.target;
+    if (eventTarget.id?.includes(id)) {
+      const newIndex = Number(eventTarget.id.split('-')[1]);
+      setActiveIndex(newIndex);
+      handleChange(newIndex);
+    }
+  };
+
+  const handleKeyDown = (event: React.KeyboardEvent) => {
+    const { which } = event;
+    let newIndex: number = activeIndex;
+
+    switch (which) {
+      case left: {
+        newIndex = activeIndex - 1;
+
+        // circularity
+        if (newIndex === -1) {
+          newIndex = tabCount - 1;
+        }
+        setActiveIndex(newIndex);
+        handleChange(newIndex);
+        break;
+      }
+      case right: {
+        newIndex = activeIndex + 1;
+
+        // circularity
+        if (newIndex === tabCount) {
+          newIndex = 0;
+        }
+        setActiveIndex(newIndex);
+        handleChange(newIndex);
+        break;
+      }
+      case home: {
+        newIndex = 0;
+        setActiveIndex(newIndex);
+        handleChange(newIndex);
+        break;
+      }
+      case end: {
+        newIndex = tabCount - 1;
+        setActiveIndex(newIndex);
+        handleChange(newIndex);
+        break;
+      }
+    }
+  };
+
+  const tabComponents = tabs.map((child, index) => {
+    const { ...other } = (child as React.ReactElement<any>).props;
+    const selected = index === value;
+
+    const config = {
+      id: `${id}-${index}`,
+      className: classNames('Tab', {
+        'Tab--active': selected
+      }),
+      tabIndex: index === value ? 0 : -1,
+      ['aria-controls']: `${id}-panel-${index}`,
+      ['aria-selected']: selected,
+      ref: index === value ? focusedTabRef : null,
+      ...other
+    };
+
+    return React.cloneElement(child as React.ReactElement<any>, config);
+  });
+
+  const tabPanelComponents = panels.map((child, index) => {
+    const { className, ...other } = (child as React.ReactElement<any>).props;
+    const panelId = `${id}-panel-${index}`;
+    return React.cloneElement(child as React.ReactElement<any>, {
+      id: panelId,
+      ['aria-labelledby']: `${id}-${index}`,
+      className: classNames('TabPanel', className, {
+        'TabPanel--hidden': value !== index
+      }),
+      ...other
+    });
+  });
+
+  useDidUpdate(() => {
+    focusedTabRef.current?.focus();
+  }, [value]);
+
+  return (
+    <div
+      className={classNames('Tabs', className, {
+        'Tabs--thin': thin
+      })}
+    >
+      <ul
+        role="tablist"
+        className="Tablist"
+        aria-label={ariaLabelForTablist || 'Tablist'}
```

Comment:
We should not specify a default here and instead require the user to provide it.
```suggestion
        aria-label={ariaLabelForTablist}
```